### PR TITLE
FINDO-8113 Analytics_Mail_Report

### PR DESCRIPTION
**Pull request by snuzifindo**

### CODE REVIEW

File: gchart/gChart.php [hunk 1]
```diff
@@ -108,10 +137,13 @@ protected function encodeData($data, $separator, $encodigData = '')
         {
             $data = $this->extendedEncodeData($data);
             $separator = '';
-        } else if ($encodigData == 't')
-        {
-            $data = $this->textEncodeData($data);
         }
+
+        /*
+         * image-charts.com needs the real value of data, not like Google that wants value 1-100.
```
Comment (TheKeymaster) [inline gchart/gChart.php:145]: `//`

File: gchart/gChart.php [hunk 2]
```diff
@@ -585,9 +618,9 @@ public function addAxisLabelPositions($axisIndex, $labelPositions)
      * @param $startVal Integer A number, definig the low value for the data set. Usually, it is the same as $startVal in addAxisRange
      * @param $endVal Integer A number, definig the high value for the data set. Usually, it is the same as $endVal in addAxisRange
```
Comment (TheKeymaster) [inline gchart/gChart.php:621]: Please remove the parameters from the docblock above and describe `$value`.

File: gchart/gChart.php [hunk 3]
```diff
@@ -718,7 +751,31 @@ public function getUrl()
...
+    }
+
+    /**
+     * @brief The URL will be signed for enterprise version of image-charts
+     * @param $url String The url to which the signiture will be appended.
+     * @return string Url with signiture in query parameter.
+     */
+    private function setSigniture($url){
+
+        if( !$this->getAccountID()|| !$this->getSecretKey() ){
```
Comment (TheKeymaster) [inline gchart/gChart.php]: Fix spaces.

File: gchart/gChart.php [hunk 4]
```diff
@@ -718,7 +751,31 @@ public function getUrl()
...
+            throw new \Exception('Account ID and Secret Key are required for enterprise version');
+        }
+
+        $rawQuerystring = http_build_query($this->chart );
```
Comment (TheKeymaster) [inline gchart/gChart.php]: One space too much.

File: gchart/gChart.php [hunk 5]
```diff
@@ -718,7 +751,31 @@ public function getUrl()
...
+        $signature = hash_hmac('sha256', $rawQuerystring,  $this->getSecretKey() );
```
Comment (TheKeymaster) [inline gchart/gChart.php]: space.

File: gchart/gChart.php [hunk 6]
```diff
@@ -718,7 +751,31 @@ public function getUrl()
...
+
+        return $url.'&ichm=' . $signature;
```
Comment (TheKeymaster) [inline gchart/gChart.php]: Spaces in between `.`.

**Review by TheKeymaster — CHANGES_REQUESTED** [review]

File: gchart/gChart.php [hunk 7]
```diff
@@ -718,7 +753,31 @@ public function getUrl()
...
+
+        return $fullUrl;
+    }
+
+    /**
+     * @brief The URL will be signed for enterprise version of image-charts
+     * @param $url String The url to which the signiture will be appended.
+     * @return string Url with signiture in query parameter.
+     */
+    private function setSigniture($url){
```
Comment (dblaichinger) [inline gchart/gChart.php]: `setSignature`

File: gchart/gChart.php (hunk 6 above)
Comment (dblaichinger) [reply gchart/gChart.php]: Consistently add spaces before all `.` concatenations.

**Review by dblaichinger — APPROVED** [review]

**Pull request merged**